src/pages/sketch/embed/sketchviewembed.js

```diff
@@ -0,0 +1,61 @@
+import React from "react"
+import { withRouter } from "react-router"
+import SketchInfo from "../../../components/sketch/sketch-info"
+import SketchViewer from "../../../components/sketch/sketch-viewer"
+import "./sketchviewembed.scss"
+import Loader from "../../../components/ui/loader"
+import AssetsAPI from "../../../api/assets"
+
+class SketchViewEmbed extends React.Component {
+  constructor(props) {
+    const { id, userid } = props.match.params
+    super(props)
+    const params = this.parseQuery(props.location.search)
+    this.state = { loading: true, id, userid, params }
+    this.getAsset()
+  }
+
+  parseQuery = (queryString) => {
+  var query = {};
+  var pairs = (queryString[0] === '?' ? queryString.substr(1) : queryString).split('&');
+  for (var i = 0; i < pairs.length; i++) {
+    var pair = pairs[i].split('=');
+    query[decodeURIComponent(pair[0])] = decodeURIComponent(pair[1] || '');
+    if (Number(query[decodeURIComponent(pair[0])])) {
+      query[decodeURIComponent(pair[0])] = Number(query[decodeURIComponent(pair[0])])
+    }
+    if (query[decodeURIComponent(pair[0])] == "true") {
+      query[decodeURIComponent(pair[0])] = true
+    }
+    if (query[decodeURIComponent(pair[0])] == "false") {
+      query[decodeURIComponent(pair[0])] = false
+    }
+  }
+  return query;
+}
```

Comment:
I'm pretty sure there's an easier way of decoding URL components than splitting the string like this, https://developer.chrome.com/blog/urlsearchparams/ URLSearchParams is probably better.

Reply:
ohhh I never knew of this API, super neat I'll make the change and resubmit !